packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -1038,6 +1038,9 @@ class TextPainter {
   /// {@endtemplate}
   TextRange getWordBoundary(TextPosition position) {
     assert(_debugAssertTextLayoutIsValid);
+    if (position.affinity == TextAffinity.upstream) {
```

Comment:
This should be in the ui.text, putting it here just for now, will spawn another pr and remove this before merge